packages/nbdime/src/common/mergeview.ts

```diff
@@ -295,21 +295,21 @@ const removeGutterMarkerEffect = StateEffect.define<{ type: string }>({
 class MergeMarker extends GutterMarker {
   constructor(options: { symbol: string; className: string; block: boolean }) {
     super();
-    this._symbol = options.symbol;
-    this._className = options.className;
-    this._block = options.block;
-  }
-  get isBlock() {
-    return this._block;
+    this.symbol = options.symbol;
+    this.className = options.className;
+    this.block = options.block;
   }
   toDOM() {
-    let pickerMarker = elt('div', this._symbol);
-    pickerMarker.className = this._className;
+    let pickerMarker = elt('div', this.symbol);
+    pickerMarker.className = this.className;
     return pickerMarker;
   }
-  private _symbol: string;
-  private _className: string;
-  private _block: boolean;
+  eq(other: GutterMarker): boolean {
+    return other === this;
```

Comment:
Should it check for equality of symbol/className/block instead?

Reply:
I wondered about it, but we are using static instances of the markers here, so I figured it would be simpler with an identity check instead. I was actually surprised I had to write this, as the [docs for `RangeValue`](https://codemirror.net/docs/ref/#state.RangeValue.eq) says:

> [...] Used when comparing rangesets. The default implementation compares by identity. [...]

`GutterMarker` extends `RangeValue`, and has its own docs for `eq` which doesn't say anything about its defaults, but in the code it always returns false:

https://github.com/codemirror/view/blob/cf5e5a9e2f349a3ca421107e0502028842ea526b/src/gutter.ts#L19

Reply:
That code also has some other stuff like an internal `compare`, and a function `sameMarkers`, that both do a bit more...